src/ir_Sharp.cpp

```diff
@@ -326,9 +329,16 @@ void IRSharpAc::setModel(const sharp_ac_remote_model_t model) {
 /// @param[in] raw Try to determine the model from the raw code only.
 /// @return The enum of the compatible model.
 sharp_ac_remote_model_t IRSharpAc::getModel(const bool raw) const {
-  if (raw)
-    return (_.A705 && _.Model) ? sharp_ac_remote_model_t::A705
-                               : sharp_ac_remote_model_t::A907;
+  if (raw) {
+    if (_.Model2) {
+      if (_.Model)
+        return sharp_ac_remote_model_t::A705;
+      else
+        return sharp_ac_remote_model_t::A903;
+    } else {
+      return sharp_ac_remote_model_t::A907;
+    }
```

Comment:
Would a switch case be cleaner, and fewer instructions?

Reply:
I thought about it too. At the moment, I think this is smaller, but yes, if we add more models etc, then this will gravitate to a switch clause.

Again, thanks for the comments and review.